### PR TITLE
[SPARK-49498][SQL][TESTS] Check differences between SR_AI and SR_Latn_AI collations

### PR DESCRIPTION
**Pull request by ilicmarkodb**
### What changes were proposed in this pull request?
Added test to check difference between SR_AI and SR_Latn_AI collations.

### Why are the changes needed?
Better testing.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Test added to `CollationSuite.scala`

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala [hunk 1]
```diff
@@ -33,7 +35,7 @@ import org.apache.spark.sql.execution.adaptive.AdaptiveSparkPlanHelper
 import org.apache.spark.sql.execution.aggregate.{HashAggregateExec, ObjectHashAggregateExec}
 import org.apache.spark.sql.execution.columnar.InMemoryTableScanExec
 import org.apache.spark.sql.execution.joins._
-import org.apache.spark.sql.internal.{SqlApiConf, SQLConf}
+import org.apache.spark.sql.internal.{SQLConf, SqlApiConf}
```
Comment (stefankandic) [inline sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala:38]: unneeded change

File: sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala [hunk 2]
```diff
@@ -186,6 +188,50 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
     }
   }
 
+  test("check difference betweeen SR_AI and SR_Latn_AI collations") {
+    // scalastyle:off nonascii
+    // SR_AI collation
+    var collationName = "SR_AI"
+    var builder = new ULocale.Builder();
```
Comment (stefankandic) [inline sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala:195]: We should still test spark's and not directly the ICU behavior, even though we use ICU under the hood

File: sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala [hunk 3]
```diff
@@ -186,6 +188,50 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
...
+
+    // SR_Latn_AI collation
+    collationName = "SR_Latn_AI"
+    builder = new ULocale.Builder();
+    builder.setLocale(new ULocale(collationName))
+    builder.setUnicodeLocaleKeyword("ks", "level1")
+    builder.setUnicodeLocaleKeyword("kc", "true")
+    collator = Collator.getInstance(builder.build())
+
+    assert(collator.compare("c", "ć") != 0)
```
Comment (stefankandic) [inline sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala:212]: we can probably put this inside a loop to avoid repetitive calls

**Comment by HyukjinKwon** [comment]
Mind filing a JIRA please? See also https://spark.apache.org/contributing.html

**Comment by ilicmarkodb** [comment]
@cloud-fan can you please review?

File: sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala [hunk 4]
```diff
@@ -188,6 +188,30 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
...
+      ("C", "Ć"),
+      ("C", "Č"),
+      ("Ć", "Č"),
+      ("s", "š"),
+      ("S", "Š"),
+      ("z", "ž"),
+      ("Z", "Ž")
+    ).foreach {
+      case (c1, c2) =>
+        checkAnswer(sql(s"SELECT '$c1' = '$c2' COLLATE SR_Latn_AI"), Row(false))
```
Comment (stefankandic) [inline sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala:210]: can we also check the result for `sr_ai` here? It's more readable than comparing multiple characters at a time like you did above

Comment (cloud-fan) [reply sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala:210]: +1

**Review by MaxGekk — APPROVED** [review]
@ilicmarkodb Please, add the `[TESTS]` tag and make PR's title shorter like:
```
Check differences between SR_AI and SR_Latn_AI collations
```

**Comment by ilicmarkodb** [comment]
@MaxGekk can you merge?

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @ilicmarkodb and @cloud-fan @stefankandic for review.

**Pull request closed without merging**